### PR TITLE
Task creater ui

### PR DESCRIPTION
**Pull request by Shilpi75**
Changed task creator ui
Modified code accordingly

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java [hunk 1]
```diff
@@ -49,35 +43,28 @@
...
-            finish();
-        });
+
     }
 
     /**
      * Shows the empty view if numLocations == 0
      *
-     * @param numLocations the number of locations being shown.
+     * @param numLocations the number of mLocations being shown.
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: Any specific reason for changing it to mLocations everywhere?

File: app/src/main/res/drawable/button_background.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (vermayash8) [inline app/src/main/res/drawable/button_background.xml]: Please avoid names like `button_background` instead use `save_button_background`. Change in other drawables too.

File: app/src/main/res/layout/content_essentials.xml [hunk 3]
```diff
@@ -0,0 +1,202 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (vermayash8) [inline app/src/main/res/layout/content_essentials.xml]: `content_essentials` isn't a descriptive name, please use `creator_essential_contents` or `content_creator_essentials`. Please do the same for others too.

File: app/src/main/res/layout/creator_main_layout.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (vermayash8) [inline app/src/main/res/layout/creator_main_layout.xml]: What's this layout file for?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java [hunk 5]
```diff
@@ -56,12 +63,6 @@
 import app.tasknearby.yashcreations.com.tasknearby.utils.WeekdayCodeUtils;
 import app.tasknearby.yashcreations.com.tasknearby.utils.firebase.AnalyticsConstants;
 
-/**
- * Creates a new task and also responsible for editing an old one. For editing, we need to use
- * the getEditModeIntent() method to get the starting intent.
- *
- * @author vermayash8
- */
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: Didn't like it?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java [hunk 6]
```diff
@@ -141,163 +152,215 @@ public static Intent getEditModeIntent(Context context, long taskId) {
...
     /**
      * Sets the toolbar as actionBar and also sets the up button.
      */
     private void setActionBar() {
-        Toolbar toolbar = findViewById(R.id.toolbar);
-        setSupportActionBar(toolbar);
         ActionBar actionBar = getSupportActionBar();
         if (null != actionBar) {
+            actionBar.setElevation(0);
+            actionBar.setTitle(R.string.title_add_new_task);
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: No need if you set label in manifest <- Ideal.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java [hunk 7]
```diff
@@ -437,45 +594,36 @@ private void onSavedPlacesSuccess(Intent data) {
             return;
         }
         long locationId = data.getLongExtra(SavedPlacesActivity.EXTRA_LOCATION_ID, -1);
+        Log.e("shilpi", locationId + "");
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: (nit): Remove

File: app/src/main/res/layout/content_attachment.xml [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        android:id="@+id/icon_note"
+        android:layout_width="wrap_content"
+        android:layout_height="24dp"
+        android:layout_marginTop="16dp"
+        app:layout_constraintStart_toStartOf="@+id/guide_ver_16dp"
+        app:layout_constraintTop_toTopOf="@+id/textInputLayout3"
+        app:srcCompat="@drawable/ic_round_note_add_24px" />
+
+    <LinearLayout
+        android:id="@+id/layout_select_image"
```
Comment (vermayash8) [inline app/src/main/res/layout/content_attachment.xml]: Note: There are no ripples at lot of places. Please add ripples to: 
* Select Location (Place Picker)
* Ring Alarm
* Select Image
* Other places you can think of.

File: app/build.gradle [hunk 9]
```diff
@@ -67,11 +67,12 @@ dependencies {
     implementation "com.google.firebase:firebase-core:$playServicesVersion"
     implementation "com.google.firebase:firebase-perf:$playServicesVersion"
     implementation "com.google.firebase:firebase-messaging:$playServicesVersion"
-    
+
     implementation "com.crashlytics.sdk.android:crashlytics:$crashlyticsVersion"
 
     // Room
     implementation "android.arch.persistence.room:runtime:$archRoomVersion"
+    implementation 'com.android.support.constraint:constraint-layout:1.1.0'
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/build.gradle]: Yeah! Sorry I forgot this. Done

File: app/src/main/AndroidManifest.xml [hunk 10]
```diff
@@ -88,11 +83,14 @@
...
             android:theme="@style/AppTheme.NoActionBar" />
-
         <activity
             android:name=".OnboardingActivity"
             android:exported="true"
             android:theme="@style/AppTheme.NoActionBar" />
+        <activity android:name=".TaskCreatorActivity"
+            android:exported="true"
+            android:parentActivityName=".MainActivity"
+            android:theme="@style/AppTheme"/>
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/AndroidManifest.xml]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java [hunk 11]
```diff
@@ -49,35 +43,28 @@
...
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_saved_places);
         getSupportActionBar().setDisplayShowHomeEnabled(true);
-        // Get the list of saved locations.
-        mTaskRepository = new TaskRepository(getApplicationContext());
+
+        mTaskRepository = new TaskRepository(this);
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java (hunk 1 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/SavedPlacesActivity.java]: No, actually I was trying something different. So I refactored the list as a class member, due to which it got changed here as well.

File: app/src/main/res/drawable/button_background.xml (hunk 2 above)
Comment (Shilpi75) [reply app/src/main/res/drawable/button_background.xml]: Done

File: app/src/main/res/layout/content_essentials.xml (hunk 3 above)
Comment (Shilpi75) [reply app/src/main/res/layout/content_essentials.xml]: Done

File: app/src/main/res/layout/creator_main_layout.xml (hunk 4 above)
Comment (Shilpi75) [reply app/src/main/res/layout/creator_main_layout.xml]: I didn't add this

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java (hunk 5 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: So got missed

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java (hunk 6 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: yes, got it

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java (hunk 7 above)
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/TaskCreatorActivity.java]: done

File: app/src/main/res/layout/content_attachment.xml (hunk 8 above)
Comment (Shilpi75) [reply app/src/main/res/layout/content_attachment.xml]: okay

**Pull request merged**